extensions/amp-story/1.0/amp-story-reaction.js

```diff
@@ -516,58 +481,46 @@ export class AmpStoryReaction extends AMP.BaseElement {
    *    ...
    *  ]
    * }
-   * @param {ReactionResponseType|undefined} response
+   * @param {ReactionResponseType} response
    * @private
    */
   handleSuccessfulDataRetrieval_(response) {
-    if (!(response && 'data' in response)) {
+    if (!(response && 'options' in response)) {
       dev().error(
         TAG,
-        `Invalid reaction response, expected { data: ReactionResponseType, ...} but received ${response}`
+        `Invalid reaction response, expected { data: Array<ReactionOptionType>, ...} but received ${response}`
       );
       return;
-    }
-
-    this.responseData_ = response.data;
-
-    this.hasUserSelection_ = this.responseData_.hasUserResponded;
-    if (this.hasUserSelection_) {
-      this.updateReactionOnDataRetrieval_(response.data);
+    } else {
+      this.updateComponentOnDataRetrieval_(
+        response['options'].splice(0, this.optionElements_.length)
+      );
     }
   }
 
   /**
-   * Updates the quiz to reflect the state of the remote data.
-   * @param {!ReactionResponseType} data
+   * Updates the component to reflect the state of the remote data.
+   * * Calls updateOptionPercentages and updateToPostSelectionState.
```

Comment:
Nit: unneeded comment
These are two private methods, we don't want to have to update the comments if we refactor the method or the private methods names